tests/unit_tests/job_runner/test_job.py

```diff
@@ -33,6 +33,47 @@ def test_run_with_process_failing(
         next(run)
 
 
+@pytest.mark.usefixtures("use_tmpdir")
+def test_memory_usage_counts_grandchildren():
+    scriptname = "recursive_memory_hog.py"
+    with open(scriptname, "w", encoding="utf-8") as script:
+        script.write(
+            """#!/usr/bin/env python
+import os
+import sys
+import time
+
+counter = int(sys.argv[1])
+numbers = list(range(int(1e6)))
+if counter > 0:
+    parent = os.fork()
+    if not parent:
+        os.execv(sys.argv[0], [sys.argv[0], str(counter - 1)])
+time.sleep(0.3)"""  # Too low sleep will make the test faster but flaky
```

Comment:
Also wondering if `sleep(0.3)` is enough?

Reply:
It seems enough. Making it too high will make the test slower so that we don't want either.